public/pages/Findings/components/FindingDetailsFlyout.tsx

```diff
@@ -300,97 +318,127 @@ export default class FindingDetailsFlyout extends Component<
     return patternId;
   };
 
-  renderFindingDocuments(isDocumentLoading: boolean) {
-    const {
-      finding: { index, document_list, related_doc_ids },
-    } = this.props;
-    const documents = document_list;
-    const docId = related_doc_ids[0];
-    const matchedDocuments = documents.filter((doc) => doc.id === docId);
-    const document = matchedDocuments.length > 0 ? matchedDocuments[0].document : '';
+  toggleDocumentDetails(item: FindingDocumentItem) {
+    const docIdToExpandedRowMapValues = { ...this.state.docIdToExpandedRowMap };
     let formattedDocument = '';
     try {
-      formattedDocument = document ? JSON.stringify(JSON.parse(document), null, 2) : '';
+      formattedDocument = document ? JSON.stringify(JSON.parse(item.document), null, 2) : '';
     } catch {
       // no-op
     }
 
-    const { indexPatternId } = this.state;
+    if (docIdToExpandedRowMapValues[item.id]) {
+      delete docIdToExpandedRowMapValues[item.id];
+    } else {
+      docIdToExpandedRowMapValues[item.id] = (
+        <EuiFormRow fullWidth={true}>
+          <EuiCodeBlock
+            language="json"
+            isCopyable
+            data-test-subj={`finding-details-flyout-rule-document-${item.itemIdx}`}
+          >
+            {formattedDocument}
+          </EuiCodeBlock>
+        </EuiFormRow>
+      );
+    }
+
+    this.setState({ docIdToExpandedRowMap: docIdToExpandedRowMapValues });
+  }
 
-    return document ? (
-      <>
-        <EuiFlexGroup justifyContent="spaceBetween">
-          <EuiFlexItem>
-            <EuiTitle size={'s'}>
-              <h3>Documents</h3>
-            </EuiTitle>
-          </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiButton
-              isLoading={isDocumentLoading}
+  renderFindingDocuments() {
+    const {
+      finding: { index, document_list, related_doc_ids },
+    } = this.props;
+    const { indexPatternId, docIdToExpandedRowMap } = this.state;
+    const relatedDocIdsSet = new Set(related_doc_ids);
+    const relatedDocuments: FindingDocumentItem[] = [];
+    document_list.forEach((documentInfo) => {
+      if (documentInfo.found && relatedDocIdsSet.has(documentInfo.id)) {
+        relatedDocuments.push({ ...documentInfo, itemIdx: relatedDocuments.length });
+      }
+    });
+
+    if (relatedDocuments.length === 0) {
+      return (
+        <>
+          <EuiTitle size={'s'}>
+            <h3>Documents</h3>
+          </EuiTitle>
+          <EuiSpacer />
+          <EuiEmptyPrompt
+            iconType="alert"
+            iconColor="danger"
+            title={<h2>Document not found</h2>}
+            body={<p>The document that generated this finding could not be loaded.</p>}
```

Comment:
For the future, it could be helpful to provide some common reasons for why the document isn't available. E.g.,
1. Was the document deleted?
2. Was the document reindexed elsewhere?
3. Etc.

Reply:
Knowing the exact reason would be tricky